### ISSUE TITLE
Speed Issues & Stuttering in L/R Frame Alternating Output Mode

### ISSUE THREAD

**Issue by jongames** [marlam/bino#16]
First, I wanted to thank you for making such a useful program and keeping it updated even when 3D technology is not being produced much anymore, while also keeping it open source.  I have used it over the past 7 years.  I recently upgraded from an older system from 2015 that was able to do opengl stereo output mode which allowed my gpu to output a frame packed HDMI signal which always worked perfectly with my 3D projector.  My new system is an RTX 3080 which has no support for the opengl quad buffering.  I found that my projector supports another mode which I had never used before called frame sequential and was happy that I may continue watching 3D movies but I had an issue.  

Since the movie I tried is in 23.97 FPS, as most are, I decided to set my display to 47.95 FPS using a custom display mode and then use the frame alternating mode both in the projector and in Bino to send the 3D data to the projector at full 1080p23.97 resolution.  This works which I am happy about, but the video stutters randomly (both in linux and windows).  The sources I tried are a Full Top over Bottom mkv so 1920x2160 resolution and a Half Side by Side MKV 1920x1080.  
Along with stutters sometimes the parallax jumps back and forth randomly only within the frame alternating output mode. Both of these issues ruin the 3D effect and may cause you to get sick sadly.  I tried many settings such as forcing VSync, triple buffering, & more through the NVIDIA control panel for Bino and also other display refresh rates (60Hz and 24Hz) but my projector does not support 1080p at anything higher than 60fps it seems.  

Another thing I noticed which I suspect may be related to the stuttering noted above is that the speed of the movie (in any output mode in Bino) depends on the refresh rate of the monitor!  I got a 165 Hz monitor recently with my new PC and on that monitor, the speed of the video is VERY fast both in Windows Bino v1.6.8 and Linux Bino v2.0.  Pausing and resuming seems to change the speed to a more normal rate but changing focus to another window on my computer messes up the speed again sometimes.  

Please tell me if you need any more details or have suggestions for me to try.

**Comment by marlam** [comment]
1. The alternating output mode is unreliable; it is just a poor replacement for true OpenGL stereo. If one output frame is missed, then left and right view swap, which ruins the experience. This mode only works if you can guarantee that your system can handle input and output without hiccups. But Bino cannot know if that's the case. That's why OpenGL stereo mode exists: then these things are handled on system level.
2. The playback speed depending on the output frequency is something that I cannot reproduce. Bino 1.6 synchronizes time to the audio output; I see no way how this can happen. In Bino 2.0, that would be a Qt bug, since Bino does not implement its own time synchronization anymore. 

So in summary, there's nothing that can be done about alternating mode being unreliable, and the playback speed issue should be debugged with Qt. Try their video player example and see if it has the same problem.

**Comment by jongames** [comment]
Thank you for your response.  I actually ended up trying to use Stereoscopic Player which is not open source these past few days.  Apparently it had the same issues!  So you are correct that this does not seem to be your program specifically.  Now I found an application that hacks the NVIDIA drivers to re-enable 3D HDMI frame packing output called 3D Fix Manager.  This works with enabling my projector to output 3D automatically using directX stereo, but has the same issue as the frame alternating in Bino: the eyes keep swapping and stuttering which makes you sick.  

After much trial and error with NVIDIA and Windows settings, I actually got the issue to mostly go away in directX stereo mode in that other player!  For reasons beyond my comprehension, setting the program to use "Power Saving" graphics mode through Windows actually made it run substantially smoother along with forcing some nvidia driver settings on that application: 
![image](https://user-images.githubusercontent.com/18472148/214142414-06f4c62b-726e-4f5c-a011-f3bbcd70d6ae.png)

I have not had a chance to try Bino again using those tweaks that allowed the other player to work but I will do so this week and report back.  I agree that it is extremely weird that the playback is changing speed with the output frequency, but based on everything I am experiencing, it seems to be a Windows 10+11 and/or NVIDIA driver issue with their newer cards and drivers.  The audio also was sped up with the video.
